ledger/types/position.go

```diff
@@ -41,13 +52,35 @@ func (p *Position) MarshalBinary() ([]byte, error) {
 }
 
 func (p *Position) UnmarshalBinary(data []byte) error {
-	return json.Unmarshal(data, p)
+	err := json.Unmarshal(data, p)
+	p.Init() // Init must be called after json unmarshaling
+	return err
 }
 
 // Init sets a position's original field to its current json representation. No-op if already set.
 // For a position object to be valid, this must be called after json unmarshaling.
 func (p *Position) Init() {
-	if p != nil && p.original == "" {
+	// Nil balances should not be allowed
+	if p.Owned == nil {
+		p.Owned = EmptyBalances()
+	}
+	if p.Locked == nil {
+		p.Locked = EmptyBalances()
+	}
+	if p.Supplied == nil {
+		p.Supplied = EmptyBalances()
+	}
+	if p.SSEQ == nil {
+		p.SSEQ = EmptyBalances()
+	}
+	if p.Inactive == nil {
+		p.Inactive = EmptyBalances()
+	}
+	if p.InterestSources == nil {
+		p.InterestSources = EmptyBalances()
+	}
+	// Store original state. No-op if already stored.
+	if p.original == "" {
```

Comment:
The package compatible with 1.0 is `"github.com/goccy/go-json"`

Reply:
`json.Marshal(p)` below is using that package, so we're good